pkg/kubernetes/pod/v1alpha1/pod.go

```diff
@@ -142,3 +142,32 @@ func IsNil() Predicate {
 func (p *Pod) GetAPIObject() *corev1.Pod {
 	return p.object
 }
+
+// FromList created a PodList with provided api podlist
+func FromList(pods *corev1.PodList, predicates ...Predicate) *PodList {
+	pl := ListBuilderForAPIList(pods).
+		WithFilter(predicates...).
+		List()
+	return pl
+}
+
+// GetNodeNames returns the nodes on which pods are running
```

Comment:
// GetScheduledNodes returns the nodes on which pods are scheduled

Reply:
Done